src/parse.c

```diff
@@ -5536,13 +5536,17 @@ static void p_formal_parameter_list(tree_t decl, type_t type)
       return;   // Was parse error
 
    tree_t p0 = tree_port(decl, 0);
-   type_add_param(type, tree_type(p0));
+   if (tree_has_type(p0))
+      type_add_param(type, tree_type(p0));
 
    if (tree_has_value(p0))
       tree_set_flag(decl, TREE_F_CALL_NO_ARGS);
```

Comment:
Maybe we can merge this into the loop below to avoid duplicating the check above. E.g.

```c
if (i == 0 && tree_has_value(p))
   tree_set_flag(decl, TREE_F_CALL_NO_ARGS);
```